core/typedefs.h

```diff
@@ -139,9 +139,22 @@ inline void __swap_tmpl(T &x, T &y) {
 }
 #endif // SWAP
 
+#if defined(__GNUC__) || defined(__clang__)
+/* GCC, Clang */
+#if defined(__i386__) || defined(__ILP32__) || (defined(__SIZEOF_POINTER__) && (__SIZEOF_POINTER__ == 4))
+#define IS_32_BIT
+#endif
+#elif defined(_MSC_VER)
+/* Microsoft Visual C++ */
+#if defined(_M_IX86)
+#define IS_32_BIT
+#endif
+#endif
```

Comment:
Sounds like a pretty old stackoverflow answer :)

Needs thorough review to do it right. I doubt we care about `__ILP32__`, but we do care about 32-bit ARM.

CC @bruvzg

Reply:
I checked it and it should be alright, basically anything GCC/LLVM uses __SIZEOF_POINTER__ anyway, so we are safe (and this also contemplates emscripten).

Reply:
32-bit ARM is still supported for Android (not sure if we do official builds, but the option is still there) as well as for Windows (completely useless in practice).

Reply:
WebAssembly is also 32-bit as of writing :slightly_smiling_face:

Reply:
The CI will tell in fact!

Reply:
Couldn't this be solved with some compile-time magic? Some SFINAE, or a more modern approach, involving pointer size and type aliasing.

Reply:
As I said, anything other than Windows is either GCC or LLVM which uses __SIZEOF_POINTER__, so this is already solved problem.